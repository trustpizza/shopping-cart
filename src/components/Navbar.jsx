import { Link } from "react-router-dom";
import PropTypes from "prop-types";

const Navbar = ({ cartCount = 0 }) => {
  return (
    <nav>
      <ul>
        <li>
          <Link to="/">Home</Link>
        </li>
        <li>
          <Link to="/shop">Store</Link>
        </li>
      </ul>
      <div data-testid="cart">
        {cartCount}
      </div>
    </nav>
  );
};

Navbar.propTypes = {
  cartCount: PropTypes.number, // Ensures cartCount is a number
};

export default Navbar;
