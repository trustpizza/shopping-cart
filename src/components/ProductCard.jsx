import { useState } from "react";
import PropTypes from "prop-types";

const ProductCard = ({ title, image, price, onClick }) => {
  const [quantity, setQuantity] = useState(1);

  return (
    <div className="card w-full max-w-xs bg-base-100 shadow-lg p-4">
      <img src={image} alt={title} className="w-full h-40 object-cover mb-4 rounded-lg" />
      <h5 className="text-xl font-semibold mb-2">{title}</h5>
      <span className="text-lg font-bold text-primary mb-2">Price: ${price}</span>
      
      <div className="flex items-center gap-2 mb-4">
        <input
          type="number"
          min="1"
          value={quantity}
          onChange={(e) => setQuantity(Number(e.target.value))}
          aria-label="Quantity"
          className="input input-bordered w-16"
        />
        <span className="text-sm">Qty</span>
      </div>
      
      <button
        type="button"
        onClick={() => onClick(quantity)}
        className="btn btn-primary w-full"
      >
        Add to Cart
      </button>
    </div>
  );
};

ProductCard.propTypes = {
  title: PropTypes.string.isRequired,  // Ensures title is a required string
  image: PropTypes.string.isRequired,  // Ensures image is a required string (URL)
  price: PropTypes.number.isRequired,  // Ensures price is a required number
  onClick: PropTypes.func.isRequired,  // Ensures onClick is a required function
};

export default ProductCard;
