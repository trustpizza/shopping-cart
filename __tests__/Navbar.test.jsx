import { describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import userEvent from "@testing-library/user-event";
import ProductCard from "../src/components/ProductCard";
import Navbar from "../src/components/Navbar";

describe("Navbar Component", () => {  
  it("renders navbar correctly", async () => {
    render(
      <MemoryRouter>
        <Navbar cartCount={0}/>
      </MemoryRouter>
    );

    // Navigation Links
    const screenLinks = await screen.findAllByRole("link");
    // Cart Object
    const cart = await screen.findByTestId("cart");

    expect(screenLinks).toHaveLength(2);
    expect(cart).toHaveTextContent("0");
  });

  it("matches snapshot", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Navbar cartCount={0}/>
      </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });

});

// it("", () => {

// });
