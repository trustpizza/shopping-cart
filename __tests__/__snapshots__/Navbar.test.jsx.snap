// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Navbar Component > matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="navbar bg-base-100 shadow-md p-4"
  >
    <div
      class="flex-1"
    >
      <a
        class="btn btn-ghost text-xl"
        data-discover="true"
        href="/"
      >
        Home
      </a>
      <a
        class="btn btn-ghost text-xl ml-4"
        data-discover="true"
        href="/shop"
      >
        Store
      </a>
    </div>
    <div
      class="flex-none"
    >
      <div
        class="indicator"
      >
        <span
          class="indicator-item badge badge-secondary"
          data-testid="cart"
        >
          0
        </span>
        <div
          class="btn btn-square btn-ghost"
        >
          ðŸ›’
        </div>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;
